### PR TITLE
fix: only mark warning on successful build

### PR DESCRIPTION
**Pull request by klu909**
## Context

it could confusing to users when failed builds are marked with warning
## Objective

only mark successful builds
## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jithine — COMMENTED** [review]
This looks like something UI shouldn't handle directly. Should we leave the STATUS override logic to API ?

**Comment by klu909** [comment]
> This looks like something UI shouldn't handle directly. Should we leave the STATUS override logic to API ?

we will keep it this way for now, main value was warning in build detail page which you wouldn't really get when fetching builds by calling api directly

**Review by jithine — APPROVED** [review]

File: app/components/workflow-graph-d3/component.js [hunk 1]
```diff
@@ -62,7 +62,12 @@ export default Component.extend({
         // set warning status if has warning
         if (builds.length) {
           builds.forEach(build => {
-            if (build.meta && build.meta.build && build.meta.build.warning) {
+            if (
+              build.meta &&
+              build.meta.build &&
+              build.meta.build.warning &&
+              build.status === 'SUCCESS'
+            ) {
```
Comment (adong) [inline app/components/workflow-graph-d3/component.js:70]: I'd say use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

`build?.meta?.build?.warning && build.status === 'SUCCESS'`

File: app/pipeline/jobs/index/controller.js [hunk 2]
```diff
@@ -122,7 +122,12 @@ export default Controller.extend(ModelReloaderMixin, {
 
           if (nextJobDetail.builds) {
             nextJobDetail.builds.forEach(build => {
-              if (build.meta && build.meta.build && build.meta.build.warning) {
+              if (
+                build.meta &&
+                build.meta.build &&
+                build.meta.build.warning &&
+                build.status === 'SUCCESS'
+              ) {
```
Comment (adong) [inline app/pipeline/jobs/index/controller.js:130]: Same `build?.meta?.build?.warning && build.status === 'SUCCESS'`

File: app/utils/metric.js [hunk 3]
```diff
@@ -100,13 +103,38 @@ const formatMetrics = metrics => {
...
-    duration: event.duration || 0,
-    statusColor: getColor(event.status.toLowerCase())
-  }));
+  const eventsInfo = metrics.map(event => {
+    let eventStatus = event.status.toLowerCase();
+
+    // only overwrite success event with warning
+    if (event && event.builds && eventStatus === 'success') {
+      event.builds.forEach(build => {
+        if (build.meta && build.meta.build && build.meta.build.warning) {
```
Comment (adong) [inline app/utils/metric.js:112]: `if (build?.meta?.build?.warning)`

File: app/utils/metric.js [hunk 4]
```diff
@@ -100,13 +103,38 @@ const formatMetrics = metrics => {
...
+      statusColor: getColor(eventStatus)
+    };
+  });
+
+  let lastEventStatus = lastEvent.status.toLowerCase();
+
+  // only overwrite success event with warning
+  if (lastEvent && lastEvent.builds && lastEventStatus === 'success') {
+    lastEvent.builds.forEach(build => {
+      if (build.meta && build.meta.build && build.meta.build.warning) {
```
Comment (adong) [inline app/utils/metric.js:129]: Same `if (build?.meta?.build?.warning)`

File: app/utils/metric.js [hunk 5]
```diff
@@ -100,13 +103,38 @@ const formatMetrics = metrics => {
...
+    return {
+      duration: event.duration || 0,
+      statusColor: getColor(eventStatus)
+    };
+  });
+
+  let lastEventStatus = lastEvent.status.toLowerCase();
+
+  // only overwrite success event with warning
+  if (lastEvent && lastEvent.builds && lastEventStatus === 'success') {
```
Comment (adong) [inline app/utils/metric.js:127]: For consistency reasons from the current file, I would use upper-case instead.
`lastEvent?.builds && build.status === 'SUCCESS'` 

Remove 
```
let lastEventStatus = lastEvent.status.toLowerCase();
```

File: app/utils/metric.js [hunk 6]
```diff
@@ -100,13 +103,38 @@ const formatMetrics = metrics => {
...
+      if (build.meta && build.meta.build && build.meta.build.warning) {
+        lastEventStatus = 'warning';
+      }
+    });
+  }
+
   const lastEventInfo = {
     startTime: `${lastEventStartMonth}/${lastEventStartDay}/${lastEventStartYear}`,
-    statusColor: getColor(lastEvent.status.toLowerCase()),
+    statusColor: getColor(lastEventStatus),
```
Comment (adong) [inline app/utils/metric.js:137]: And keep `statusColor: getColor(lastEvent.status.toLowerCase())`

**Review by adong — CHANGES_REQUESTED** [review]
See comments for code readability improvements

**Review by adong — APPROVED** [review]

**Pull request merged**